### ISSUE TITLE
Let ACE grow the height of the contents

### ISSUE THREAD

**Issue by bertoost** [danyaPostfactum/modx-ace#4]
Love this feature of CodeMirror integration but liked the way ACE works.
Shouldn't be so hard I think to let the ACE editor grow the height the contents length is

**Comment by danyaPostfactum** [comment]
I'm thinking about it.
For now, you may get it by adding two parameters to "codearea" initialization:

plugin Ace:

var CodeArea = MODx.load({
    ...
    grow: true,
    growMax: Number.MAX_VALUE
})

**Comment by bertoost** [comment]
Thanks for looking over it.. would be great to have this feature (or at least to can enable it trought system settings). Modifying the plugin doesn't work for me.

**Comment by danyaPostfactum** [comment]
Try to clear modx cache. When page is loaded, editarea will not grow
up immediatly. You have to give focus to editarea at first.

**Comment by bertoost** [comment]
You're right. The cache was cleared by the way, but I didn't focus the area. That's working now. Would love to have this feature by default! Likely also without having to focus the field (so in case of emergency, trigger the focus event onload ;-) hehe)

**Comment by ethanpooley** [comment]
Where would those two parameters need to be added? There is a spot in manager/components/ace/assets/modx.codearea.js that looks similar, but not identical, but it didn't do anything when I added them their. Apologies - very new to ModX and plugin structure.

**Comment by bertoost** [comment]
Hi Ethan.. It should be in the plugin because this is loaded in the manager. You can find the plugin in the Elements tree.. Scroll down a bit for the code

**Comment by ethanpooley** [comment]
Ho! Nice. Thanks.

**Comment by danyaPostfactum** [comment]
I have added new feature - full window mode. Press Ctrl+F11 for entering this mode.

autoHeight is bad due to perfomance problems...

**Comment by bertoost** [comment]
I never experienced performance issues with the "grow" and "growMax" settings. I always set them when I install Ace.

You can add these two settings to the system settings for Ace component so people can disable it or change the growMax value when they do experience that.

The fullscreen mode is nice but it's totally different then the grow option. With fullscreen you even cannot view other elements or resources (when you need some ID or something), so you will need to switch continuously between fullscreen and back. Thats not cool.

**Comment by danyaPostfactum** [comment]
This issue is related to MODx. If you like grow feature, create the issue for it in modx tracker. But I don't think core developers like this feature.

**Comment by bertoost** [comment]
Why do you think this is a MODX feature?? It's inside your component right? Not the MODX code.. I have already made an update for this and send you the pull request already (including system settings and the movement of assets folder, as long MODX not supports this).

**Comment by danyaPostfactum** [comment]
> It's inside your component right? Not the MODX code

No, I'm going to offer to modx developers the way to use default code and html editors in any place (popup windows, third-party components etc). It is based on ExtJS features, so all parameters, such as grow, growMax should be defined in ext-components (panels) declarations. I hope they will merge my patch in modx-2.3 version (this is fully compatible with old extras)

**Comment by bertoost** [comment]
I think you're totally misunderstand the usage of components against MODX core code.

Every component may have his own features, may use configuration possibilities there are in Extjs without falling back on the core code. That's why it is a configuration setting, right??

Sorry but I totally do not agree this with you.

**Comment by danyaPostfactum** [comment]
In next version my plugin will only provide extjs component (by registering it as 'modx-texteditor' xtype), and there will not be any dependencies on `current panel` or `current page` or something else. Editarea will look exactly as specified by panel definition. That's why this feature should be in the core. For example, popup windows should not allow grow editarea, but element panels should.

That's why I don't want to add this feature to Ace. The second reason is that this feature is a bit expensive (i mean perfomance). But it seems this is almost doesn't make the sence (at least on modern machines)

**Comment by bertoost** [comment]
Alright; I don't get this at all.. and as far I know, windows do not have any fancy editor things (even not for WYSIWYG editor) and this should always be different then the normal panels.

Nice to see you're going to make a Ext.reg() of it.. but you always should handle this with a plugin to load your own registered component. And then; why do you decide for other what they can and what they can't do.. let them decide at themself (as in "Creative freedom!").

**Comment by adamwintle** [comment]
I've been looking for a way to make the Ace edit area grow to the full height of the browser - has there been any progress on this?

You also mentioned pressing Ctrl+F11 will enter a full-screen mode, but I can't seem to get this to work on the Mac; is it the same keyboard shortcut?

**Comment by danyaPostfactum** [comment]
For Mac Command+F12 is used now: https://github.com/danyaPostfactum/modx-ace/issues/22

**Comment by jpdevries** [comment]
> For Mac Command+F12 is used now: #22

This keyboard shortcut isn't working for me on Mac. Also, the editor is only 4 lines tall (when editing a 17 line tall chunk) regardless of the `ace.height` setting.

Ace 1.5.1 MODX 2.3.3.

![](http://j4p.us/image/44333Q413h2H/Screen%20Shot%202015-07-07%20at%209.43.53%20AM.png)

**Comment by syberknight** [comment]
is there any resolution to this yet? i just replaced CodeMirror with ACE & love the styles - but i REALLY liked the way CM auto-expanded the textarea height to fill the space. provided for a much larger code area, but still within the page area to still see & have quick access to everything else. i like the fullscreen option for "sometimes" (note: i went in to the "assets/components/ace/modx.textarea.js" & bumped the "left" css attribute from 0 to 250px to at least keep the file tree in view).

but another problem occurred when i change the theme to a dark theme... scrollbars go missing!

i've been scouring the net & the code for clues on how to fix, but have come up empty. please help. thx.

**Comment by Ibochkarev** [comment]
Hi! Auto-grow is available via the `ace.grow` system setting — see [modx-pro/modx-ace PR #3](https://github.com/modx-pro/modx-ace/pull/3) and [changelog](https://github.com/modx-pro/modx-ace/blob/master/core/components/ace/documents/changelog.txt). Active repo: [modx-pro/modx-ace](https://github.com/modx-pro/modx-ace).
